eva-accession-release-automation/gather_clustering_counts/gather_release_counts.py

```diff
@@ -0,0 +1,300 @@
+            vcf_files = glob.glob(os.path.join(assembly_dir, vcf_pattern))
+            txt_pattern = f'*_GCA_*_ids.txt.gz'
+            txt_files = glob.glob(os.path.join(assembly_dir, txt_pattern))
+            # I don't have the taxonomy to add to the bash pattern so remove the file that start with eva_ or dbsnp_
+            vcf_files = [f for f in vcf_files if 'dbsnp_' not in f and 'eva_' not in f]
+            txt_files = [f for f in txt_files if 'dbsnp_' not in f and 'eva_' not in f]
+            all_files.extend(vcf_files)
+            all_files.extend(txt_files)
+        all_files.extend(glob.glob(os.path.join(species_dir, '*_unmapped_ids.txt.gz')))
+    return all_files
+
+
+def calculate_all_logs(release_dir, output_dir, species_directories=None):
+    all_assemblies_2_species, all_species_2_assemblies = gather_assemblies_and_species_from_directory(release_dir)
+    all_sets_of_species = set()
+    # Determine the species that needs to be counted together because they share assemblies
+    species_to_search = species_directories
+    if not species_to_search:
+        species_to_search = all_species_2_assemblies.keys()
+    logger.info(f'Process {len(species_to_search)} species')
+    for species in species_to_search:
+        set_of_species, set_of_assemblies = find_link({species}, all_species_2_assemblies, all_assemblies_2_species)
+        all_sets_of_species.add(set_of_species)
+    logger.info(f'Aggregate species in {len(all_sets_of_species)} groups')
+    all_logs = []
+    for set_of_species in all_sets_of_species:
+        logger.info(f'Process files for {",".join(set_of_species)} groups')
+        all_logs.append(gather_count_for_set_species(release_dir, set_of_species, output_dir))
+    return all_logs
+
+
+def generate_output_tsv(dict_of_counter, output_file, header):
+    with open(output_file, 'w') as open_file:
+        open_file.write("\t".join([header, 'Metric', 'Count']) + '\n')
+        for annotation1 in dict_of_counter:
+            for annotation2 in dict_of_counter[annotation1]:
+                open_file.write("\t".join([
+                    str(annotation1), str(annotation2), str(dict_of_counter[annotation1][annotation2])
+                ]) + '\n')
+
+
+class ReleaseCounter(AppLogger):
+
+    def __init__(self, private_config_xml_file, release_version, logs):
+        self.private_config_xml_file = private_config_xml_file
+        self.release_version = release_version
+        self.all_counts_grouped = []
+        self.parse_logs(logs)
+        self.add_annotations()
+
+    @lru_cache
+    def get_taxonomy_and_scientific_name(self, species_folder):
+        query = (
+            f"select distinct c.taxonomy, t.scientific_name "
+            f"from eva_progress_tracker.clustering_release_tracker c "
+            f"join evapro.taxonomy t on c.taxonomy=t.taxonomy_id "
+            f"where release_version={self.release_version} AND release_folder_name='{species_folder}'"
```

Comment:
Hope this works with release folders with weird names or trailing underscores in them...

Reply:
I think it should. It is currently working for all folder names.